server/taskflows/hpccloud/taskflow/nwchem/__init__.py

```diff
@@ -100,30 +109,35 @@ def setup_input(task, *args, **kwargs):
     client = create_girder_client(
         task.taskflow.girder_api_url, task.taskflow.girder_token)
 
-    # Get the geometry file metadata to see if we need to import
-    geometry_file = client.getResource('file/%s' % geometry_file_id)
-    kwargs['geometryFilename'] = geometry_file['name']
+    geometry_file_id = parse('input.geometryFile.id').find(kwargs)
+    if geometry_file_id:
+        geometry_file_id = geometry_file_id[0].value
+        kwargs['geometryFileId'] = geometry_file_id
+
+        # Get the geometry file metadata to see if we need to import
+        geometry_file = client.getResource('file/%s' % geometry_file_id)
+        kwargs['geometryFilename'] = geometry_file['name']
 
     ini_file_id = kwargs['input']['nwFile']['id']
     ini_file = client.getResource('file/%s' % ini_file_id)
     kwargs['nwFilename'] = ini_file['name']
 
-    create_job.delay(*args, **kwargs)
+    return kwargs
```

Comment:
I assume since setup_input still takes and now returns kwargs that it must always be first in a taskflow chain?